python/docs/source/index.rst

```diff
@@ -47,6 +47,8 @@ With this package, you can:
 * Have a single codebase that works both with pandas (tests, smaller datasets) and with Spark (distributed datasets).
 * Switch to pandas API and PySpark API contexts easily without any overhead.
 
+Note that pandas has different behavior per its version, and pandas API on Spark tries to match the behavior of pandas 1.4.
```

Comment:
Let's add this in https://spark.apache.org/docs/latest/api/python/reference/index.html page. Also `pandas 1.4.` -> latest pandas release.

We should specify the version only in other branches.

Reply:
Sure, then can I create other ones for `branch-3.3` and `branch-3.2` separately ?

I believe that `branch-3.2` supports pandas 1.3 and `branch-3.3` supports pandas 1.4.

Reply:
https://github.com/Yikun/spark/pull/86

FYI, there are some test failed with pandas 1.4.x, I haven't got enough time to find all errors reason yet, some of thems are some panda bugs in 1.4.x (such as https://github.com/pandas-dev/pandas/issues/46589). we only tested 1.3.x with github action ci.

Reply:
I created SPARK-38819 to track the issue on `Run Pandas on Spark with Pandas 1.4.x`.

Reply:
@Yikun Thanks for the testing!! Let me help taking a look